Mobile Buy SDK/Mobile Buy SDK/Models/BUYAccountCredentials.h

```diff
@@ -25,40 +25,55 @@
 //
 
 #import <Foundation/Foundation.h>
+NS_ASSUME_NONNULL_BEGIN
 
-/**
- *   Intended for storing a collection of credential items representing individual values
- */
+extern NSString * const BUYAccountFirstNameKey;
+extern NSString * const BUYAccountLastNameKey;
+extern NSString * const BUYAccountEmailKey;
+extern NSString * const BUYAccountPasswordKey;
+extern NSString * const BUYAccountPasswordConfirmationKey;
 
 @class BUYAccountCredentialItem;
-@interface BUYAccountCredentials : NSObject
 
-NS_ASSUME_NONNULL_BEGIN
+/**
+ * Encapsulates user's credentials represented by BUYAccountCredentialItem
+ * objects.
+ */
+@interface BUYAccountCredentials : NSObject
 
-+ (BUYAccountCredentials *)credentialsWithItems:(NSArray<BUYAccountCredentialItem *> *)items;
-+ (BUYAccountCredentials *)credentialsWithItemKeys:(NSArray<NSString *> *)keys;
+@property (nonatomic, assign, readonly) NSUInteger count;
+@property (nonatomic, assign, readonly, getter=isValid) BOOL valid;
+@property (nonatomic, strong, readonly) NSDictionary *JSONRepresentation;
 
-@property (readonly) NSDictionary *JSONRepresentation;
-@property (nonatomic, readonly, getter=isValid) BOOL valid;
++ (BUYAccountCredentials *)credentials;
```

Comment:
How are values assigned in these cases if both the `BUYAccountCredentials` and `BUYAccountCredentialItem` are immutable? If go the immutable route, I imagine we'll only have on initialize (plus convenience class initializer), no?